File: backend/ms-tarefa/src/domain/entities/Tarefa.ts
export type TarefaRecorrencia = 'unica' | 'diaria' | 'semanal' | 'mensal';
export type TarefaStatus = 'pendente' | 'concluida' | 'arquivada';

export interface Tarefa {
  id: string;
  associacaoId: string;
  imovelId: string;
  proprietarioId: string;
  comodatarioId: string;
  titulo: string;
  descricao: string | null;
  recorrencia: TarefaRecorrencia;
  prazo: string | null;
  status: TarefaStatus;
  concluidaEm: string | null;
  createdAt: string;
}

export interface TarefaRow {
  id: string;
  associacao_id: string;
  imovel_id: string;
  proprietario_id: string;
  comodatario_id: string;
  titulo: string;
  descricao: string | null;
  recorrencia: TarefaRecorrencia;
  prazo: string | null;
  status: TarefaStatus;
  concluida_em: string | null;
  created_at: string;
}

export function rowToTarefa(row: TarefaRow): Tarefa {
  return {
    id: row.id,
    associacaoId: row.associacao_id,
    imovelId: row.imovel_id,
    proprietarioId: row.proprietario_id,
    comodatarioId: row.comodatario_id,
    titulo: row.titulo,
    descricao: row.descricao,
    recorrencia: row.recorrencia,
    prazo: row.prazo,
    status: row.status,
    concluidaEm: row.concluida_em,
    createdAt: row.created_at,
  };
}

File: mobile/cliente/lib/features/tarefas/domain/tarefa.dart
class Tarefa {
  const Tarefa({
    required this.id,
    required this.associacaoId,
    required this.imovelId,
    required this.proprietarioId,
    required this.comodatarioId,
    required this.titulo,
    this.descricao,
    required this.recorrencia,
    this.prazo,
    required this.status,
    required this.createdAt,
    this.concluidaEm,
  });

  final String id;
  final String associacaoId;
  final String imovelId;
  final String proprietarioId;
  final String comodatarioId;
  final String titulo;
  final String? descricao;
  final String recorrencia; // unica/diaria/semanal/mensal
  final String? prazo;
  final String status; // pendente/concluida/arquivada
  final DateTime createdAt;
  final DateTime? concluidaEm;

  bool get isPendente => status == 'pendente';

  factory Tarefa.fromJson(Map<String, dynamic> j) => Tarefa(
    id: j['id'] as String,
    associacaoId: j['associacaoId'] as String,
    imovelId: j['imovelId'] as String,
    proprietarioId: j['proprietarioId'] as String,
    comodatarioId: j['comodatarioId'] as String,
    titulo: j['titulo'] as String,
    descricao: j['descricao'] as String?,
    recorrencia: (j['recorrencia'] as String?) ?? 'unica',
    prazo: j['prazo'] as String?,
    status: (j['status'] as String?) ?? 'pendente',
    createdAt:
        DateTime.tryParse((j['createdAt'] as String?) ?? '') ?? DateTime.now(),
    concluidaEm: j['concluidaEm'] == null
        ? null
        : DateTime.tryParse(j['concluidaEm'] as String),
  );
}
